### DISCUSSION TITLE
Select particular date in calender

### DISCUSSION THREAD

**Discussion by swati1701** [mobile-dev-inc/Maestro#2216] [Q&A]
Hi,

I am using Maestro to automate leave application in mobile app. I need to select particular date in the calendar. Can someone please help me on this.

I am new to automation and using Maestro as well. 

Thanks

**Comment by portyu9** [comment]
Hi @swati1701 

Maestro does not have a universal `selectDate` command because Android/iOS/app-owned calendar widgets expose dates differently.

The best approach is to select the date through whatever **accessibility text or ID the calendar exposes**.

For an app-owned calendar, I recommend giving each date a stable identifier, for example:

```yaml
- tapOn:
    id: "date-2026-09-15"
```

For a native/system date picker, inspect it in Maestro Studio/the hierarchy and target the accessibility label it exposes. System calendars often expose something more specific than just `"15"` (for example, the full date), which is much safer because calendar views can contain duplicate day numbers.

If you only have visible day numbers, you can use relational/index selectors to disambiguate them, but the exact selector depends on the hierarchy of your calendar.

So the missing information here is which calendar you are using (Android/iOS and native vs app-owned). The stable solution is to target its accessible date representation rather than fixed screen coordinates.

Please mark this answer as accepted if it helped, thank you.
